generator/osm2type.cpp

```diff
@@ -662,38 +672,35 @@ void PreprocessElement(OsmElement * p, CalculateOriginFnT const & calcOrg)
       {"station", "light_rail", [&isLightRail] { isLightRail = true; }},
 
       {"capital", "yes", [&isCapital] { isCapital = true; }},
+
+      {"type", "multipolygon", [&isMultipolygon] { isMultipolygon = true; }},
   });
 
   if (!hasLayer && layer)
     p->AddTag("layer", layer);
 
-  // Tag 'city' is needed for correct selection of metro icons.
-  if ((isSubway || isLightRail) && p->m_type == OsmElement::EntityType::Node)
-  {
-    string const city = MatchCity(p);
-    if (!city.empty())
-      p->AddTag("city", city);
-  }
-
-  p->AddTag("psurface", DetermineSurface(p));
-
-  // Convert public_transport tags to the older schema.
-  for (auto const & tag : p->m_tags)
+  if (p->m_type == OsmElement::EntityType::Node)
   {
-    if (tag.m_key == "public_transport")
+    // Tag 'city' is needed for correct selection of metro icons.
+    if (isSubway || isLightRail)
     {
-      if (tag.m_value == "platform" && isBus)
-      {
-        if (p->m_type == OsmElement::EntityType::Node)
-          p->AddTag("highway", "bus_stop");
-      }
-      else if (tag.m_value == "stop_position" && isTram && p->m_type == OsmElement::EntityType::Node)
-      {
-        p->AddTag("railway", "tram_stop");
-      }
-      break;
+      string const city = MatchCity(p);
+      if (!city.empty())
+        p->AddTag("city", city);
     }
+
+    // Convert public_transport tags to the older schema.
+    if (isPlatform && isBus)
+      p->AddTag("highway", "bus_stop");
```

Comment:
is it possible it'd create a duplicate tag if it existed already?

Reply:
Maybe, but no problem here.